.github/workflows/test.yml

```diff
@@ -9,6 +9,23 @@ on:
       - '.gitignore'
 
 jobs:
+  tests:
+    runs-on: ubuntu-latest
+    container:
+      image: python:3.12
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt
```

Comment:
🔧 Ensure that the `requirements.txt` is up-to-date with all necessary dependencies for the tests to run smoothly.